### PR TITLE
add actions to state-badge element and state-label-badge (default Lovelace badge type)

### PR DESCRIPTION
**Pull request by iantrich**
Docs: https://github.com/home-assistant/home-assistant.io/pull/10915

### CODE REVIEW

**Comment by iantrich** [comment]
will wait for https://github.com/home-assistant/home-assistant.io/pull/10834 to be merged before adding docs

File: src/components/entity/ha-state-label-badge.ts [hunk 1]
```diff
@@ -35,6 +35,8 @@ export class HaStateLabelBadge extends LitElement {
 
   @property() public image?: string;
 
+  @property() public handleClick?: boolean;
```
Comment (bramkragten) [inline src/components/entity/ha-state-label-badge.ts]:
```
  @property({type : Boolean}) public dontHandleClick;
```

File: src/components/entity/ha-state-label-badge.ts [hunk 2]
```diff
@@ -92,12 +94,15 @@ export class HaStateLabelBadge extends LitElement {
...
   protected firstUpdated(changedProperties: PropertyValues): void {
     super.firstUpdated(changedProperties);
-    this.addEventListener("click", (ev) => {
-      ev.stopPropagation();
-      if (this.state) {
-        fireEvent(this, "hass-more-info", { entityId: this.state.entity_id });
-      }
-    });
+
+    if (this.handleClick !== false) {
```
Comment (bramkragten) [inline src/components/entity/ha-state-label-badge.ts]:
```
    if (this.dontHandleClick) {
        return;
    }
```

File: src/panels/lovelace/elements/hui-state-badge-element.ts [hunk 3]
```diff
@@ -61,9 +64,28 @@ export class HuiStateBadgeElement extends LitElement
           : this._config.title === null
           ? ""
           : this._config.title}"
+        @ha-click=${this._handleClick}
+        @ha-hold=${this._handleHold}
+        @ha-dblclick=${this._handleDblClick}
+        .longPress=${longPress({
+          hasDoubleClick: hasDoubleClick(this._config!.double_tap_action),
+        })}
+        .handleClick=${false}
```
Comment (bramkragten) [inline src/panels/lovelace/elements/hui-state-badge-element.ts]:
```suggestion
        dontHandleClick
```

**Comment by bramkragten** [comment]
How many places use `HaStateLabelBadge`? Rather have 1 logic instead of 2, either have them all handle the clicks themselves or have `HaStateLabelBadge` deal with it.

**Comment by iantrich** [comment]
> How many places use `HaStateLabelBadge`? Rather have 1 logic instead of 2, either have them all handle the clicks themselves or have `HaStateLabelBadge` deal with it.

Agreed

**Comment by iantrich** [comment]
Went with a different approach and as suggested have the parent handling click events. I realized 3/4 the way through moving ha-badges-card to Lit that it was part of States, :man_shrugging: 

In doing this I also decided to add actions to badges :tada:

**Review by bramkragten — APPROVED** [review]

**Pull request merged**